### PR TITLE
fix(toolbar): Move buttons to overflow menu when the space isn't enough

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: react/features/invite/components/InfoDialogButton.web.js [hunk 1]
```diff
@@ -179,6 +199,23 @@ class InfoDialogButton extends Component<Props, State> {
...
+     * @returns {void}
+     */
+    _onClickOverflowMenuButton() {
+        const { _dialIn, _liveStreamViewURL } = this.props;
+
+        this.props.dispatch(openDialog(InfoDialog, {
+            dialIn: _dialIn,
+            liveStreamViewURL: _liveStreamViewURL,
+            isInlineDialog: false
+        }));
```
Comment (virtuacoplenny) [inline react/features/invite/components/InfoDialogButton.web.js]: I can't add a password from within the dialog. Pressing enter dismisses the dialog.

File: react/features/invite/components/info-dialog/InfoDialog.web.js [hunk 2]
```diff
@@ -246,6 +257,20 @@ class InfoDialog extends Component<Props, State> {
...
+
+        return (
+            <Dialog
+                cancelTitleKey = 'dialog.close'
+                submitDisabled = { true }
+                titleKey = 'info.label'
+                width = 'small'>
+                { inlineDialog }
+            </Dialog>
+        );
```
Comment (virtuacoplenny) [inline react/features/invite/components/info-dialog/InfoDialog.web.js]: It makes more sense to me if this component in general were called MeetingInfo and then the consumer of MeetingInfo could dictate how to display it, be it in a Dialog or on its own. Up to you.

File: react/features/toolbox/components/web/Toolbox.js [hunk 3]
```diff
@@ -924,6 +1001,88 @@ class Toolbox extends Component<Props> {
...
+            case 'invite':
+                return (
+                    <OverflowMenuItem
+                        accessibilityLabel =
+                            { t('toolbar.accessibilityLabel.invite') }
+                        icon = 'icon-invite'
+                        key = 'invite'
+                        onClick = { this._onToolbarOpenInvite }
+                        text = { t('toolbar.invite') } />
+                );
```
Comment (virtuacoplenny) [inline react/features/toolbox/components/web/Toolbox.js]: Do you want to go ahead and add tooltips to all these using OverflowMenuItem? Right now some will have them and others won't. The perpetual-to-do of moving all buttons to use AbstractButton would have solved this but might or might not be more effort.

File: react/features/toolbox/components/web/Toolbox.js [hunk 4]
```diff
@@ -273,6 +289,8 @@ class Toolbox extends Component<Props> {
                     shortcut.helpDescription);
             }
         });
+
+        window.addEventListener('resize', this._onResize);
```
Comment (virtuacoplenny) [inline react/features/toolbox/components/web/Toolbox.js]: Not worth debouncing the callback? Probably an infrequent enough case not to but checking in anyway.

Poking around the mouseover in this file needs debouncing too.

File: react/features/toolbox/components/web/Toolbox.js [hunk 5]
```diff
@@ -941,13 +1100,88 @@ class Toolbox extends Component<Props> {
...
+        const buttonsRight = [];
+
+        const maxNumberOfButtonsPerGroup = Math.floor(
+            (
+                this.state.windowWidth
+                    - 168 // the width of the central group by design
+                    - 48 // the minimum space between the button groups
+            )
+            / 56 // the width + padding of a button
+            / 2 // devide by the number of groups(left and right group)
```
Comment (virtuacoplenny) [inline react/features/toolbox/components/web/Toolbox.js]: divide

Have you considered dynamically calculating? Might not be worth the hassle and code though.

File: react/features/toolbox/components/web/Toolbox.js [hunk 6]
```diff
@@ -941,13 +1100,88 @@ class Toolbox extends Component<Props> {
...
+        );
+
+        console.log(maxNumberOfButtonsPerGroup);
```
Comment (virtuacoplenny) [inline react/features/toolbox/components/web/Toolbox.js]: Intentionally left in?

File: react/features/toolbox/components/web/Toolbox.js [hunk 7]
```diff
@@ -941,13 +1100,88 @@ class Toolbox extends Component<Props> {
...
+        }
+
+        overflowMenuContent.splice(
+            1, 0, ...this._renderMovedButtons(movedButtons));
 
         return (
             <div className = 'toolbox-content'>
                 <div className = 'button-group-left'>
-                    { this._shouldShowButton('desktop')
+                    { buttonsLeft.indexOf('desktop') !== -1
```
Comment (virtuacoplenny) [inline react/features/toolbox/components/web/Toolbox.js]: Array.prototype.includes I think reads much better than indexOf.

File: react/features/invite/components/info-dialog/InfoDialog.web.js (hunk 2 above)
Comment (hristoterezov) [reply react/features/invite/components/info-dialog/InfoDialog.web.js]: But then we'll need 2 more classes/files for adding the Dialog and the InlineDialog wrappers, isn't it?

File: react/features/toolbox/components/web/Toolbox.js (hunk 3 above)
Comment (hristoterezov) [reply react/features/toolbox/components/web/Toolbox.js]: Even before adding this logic there were some buttons that have tooltips and others don't have. IMHO we should remove the tooltips from the buttons in the overflow menu.

I'll prefer to solve the problem for all buttons in the overflow menu trough another PR. I guess it worth discussing it internally.

File: react/features/toolbox/components/web/Toolbox.js (hunk 4 above)
Comment (hristoterezov) [reply react/features/toolbox/components/web/Toolbox.js]: Yes. I don't think it worth. Also when I tested it all looked fine...

File: react/features/toolbox/components/web/Toolbox.js (hunk 5 above)
Comment (hristoterezov) [reply react/features/toolbox/components/web/Toolbox.js]: >  dynamically calculating

What do you mean?

File: react/features/toolbox/components/web/Toolbox.js (hunk 6 above)
Comment (hristoterezov) [reply react/features/toolbox/components/web/Toolbox.js]: Will remove it!

File: react/features/toolbox/components/web/Toolbox.js (hunk 7 above)
Comment (hristoterezov) [reply react/features/toolbox/components/web/Toolbox.js]: It does! But wasn't indexOf faster? I was thinking this render could be called many times... WDYT?

If you are not worried about the performance I'll change it!

Comment (virtuacoplenny) [reply react/features/toolbox/components/web/Toolbox.js]: jsperf shows about 500 more operations a second over includes; fairly negligible and depends on chrome version. If you want to get into performance you should look at changing the buttons to a set.

File: react/features/base/toolbox/components/AbstractButton.js [hunk 8]
```diff
@@ -160,8 +160,12 @@ export default class AbstractButton<P: Props, S: *> extends Component<P, S> {
      * @returns {string}
      */
     _getIconName() {
-        return (this._isToggled() ? this.toggledIconName : this.iconName)
-            || this.iconName;
+        const { showLabel } = this.props;
```
[the remark this answers is not in the document]
Comment (hristoterezov) [reply react/features/base/toolbox/components/AbstractButton.js]: At least for now when we have label shown we change the label and icon stays the same.

[the remark this answers is not in the document]
Comment (hristoterezov) [reply react/features/base/toolbox/components/AbstractButton.js]: Did I introduced a bug? Have a missed an use case?

[the remark this answers is not in the document]
Comment (hristoterezov) [reply react/features/base/toolbox/components/AbstractButton.js]: After offline discussion with @saghul I realised that this change will brake mobile. Now this should be handled by the css and the issue should be fixed.

**Comment by saghul** [comment]
@hristoterezov sorry, here i smy belated LGTM 👍

**Pull request merged**